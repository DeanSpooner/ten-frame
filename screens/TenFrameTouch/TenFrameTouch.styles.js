import { StyleSheet } from "react-native";

export default tenFrameTouchStyles = StyleSheet.create({
  text: { color: "#fff", fontSize: 20, marginBottom: 15 },
  amount: {
    color: "#fff",
    fontSize: 80,
    marginTop: 15,
  },
});
