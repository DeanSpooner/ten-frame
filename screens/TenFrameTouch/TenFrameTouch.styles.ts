import { StyleSheet } from "react-native";

const tenFrameTouchStyles = StyleSheet.create({
  text: { color: "#fff", fontSize: 20, marginBottom: 15 },
  amount: {
    color: "#fff",
    fontSize: 80,
    marginTop: 15,
  },
});

export default tenFrameTouchStyles;
