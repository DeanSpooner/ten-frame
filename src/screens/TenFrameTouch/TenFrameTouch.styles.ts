import { StyleSheet } from "react-native";

const tenFrameTouchStyles = StyleSheet.create({
  amount: {
    color: "#fff",
    fontSize: 80,
    marginTop: 15,
  },
});

export default tenFrameTouchStyles;
