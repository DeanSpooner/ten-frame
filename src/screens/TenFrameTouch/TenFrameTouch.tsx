import React, { useState } from "react";
import { Text, View } from "react-native";
import globalStyles from "../../../Global.styles";
import TouchTable from "../../components/TouchTable/TouchTable";
import tenFrameTouchStyles from "./TenFrameTouch.styles";

const TenFrameTouch: React.FC<{}> = () => {
  const [countersTouched, setCountersTouched] = useState<number>(0);

  return (
    <View style={globalStyles.screenContainer}>
      <Text style={globalStyles.text}>
        Touch the ten frame to add counters to it.
      </Text>
      <TouchTable
        countersTouched={countersTouched}
        setCountersTouched={setCountersTouched}
      />
      <Text style={tenFrameTouchStyles.amount}>{countersTouched}</Text>
    </View>
  );
};

export default TenFrameTouch;
