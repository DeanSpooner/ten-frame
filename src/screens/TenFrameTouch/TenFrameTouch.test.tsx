import React from "react";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import TenFrameTouch from "./TenFrameTouch";

describe("<TenFrameTouch />", () => {
  const tree = renderer
    .create(<TenFrameTouch />)
    .toJSON() as ReactTestRendererJSON[];

  it("renders correctly", () => {
    expect(tree).toMatchSnapshot();
  });

  it("contains 3 direct child components", () => {
    //@ts-ignore
    expect(tree?.children.length).toBe(3);
  });
});
