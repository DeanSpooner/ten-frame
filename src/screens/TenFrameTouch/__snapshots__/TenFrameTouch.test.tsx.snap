// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TenFrameTouch /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 20,
      "minHeight": 1234,
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 41.666666666666664,
        "margin": 15,
        "textAlign": "center",
      }
    }
  >
    Touch the ten frame to add counters to it.
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 142.5,
          "justifyContent": "center",
          "opacity": 1,
          "width": "19%",
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 142.5,
          "justifyContent": "center",
          "opacity": 1,
          "width": "19%",
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 142.5,
          "justifyContent": "center",
          "opacity": 1,
          "width": "19%",
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 142.5,
          "justifyContent": "center",
          "opacity": 1,
          "width": "19%",
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 142.5,
          "justifyContent": "center",
          "opacity": 1,
          "width": "19%",
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 142.5,
          "justifyContent": "center",
          "opacity": 1,
          "width": "19%",
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 142.5,
          "justifyContent": "center",
          "opacity": 1,
          "width": "19%",
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 142.5,
          "justifyContent": "center",
          "opacity": 1,
          "width": "19%",
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 142.5,
          "justifyContent": "center",
          "opacity": 1,
          "width": "19%",
        }
      }
    >
      <View
        style=""
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderWidth": 3,
          "height": 142.5,
          "justifyContent": "center",
          "opacity": 1,
          "width": "19%",
        }
      }
    >
      <View
        style=""
      />
    </View>
  </View>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 80,
        "marginTop": 15,
      }
    }
  >
    0
  </Text>
</View>
`;
